pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go

```diff
@@ -99,6 +106,28 @@ func (fam *RemoteSecondaryForkedAlertmanager) ApplyConfig(ctx context.Context, c
 		}
 	}()
 
+	if fam.shouldFetchRemoteState {
+		wg.Wait()
+		if !fam.remote.Ready() {
+			return fmt.Errorf("remote Alertmanager not ready, can't fetch remote state")
+		}
+		// Pull and merge the remote Alertmanager state.
+		rs, err := fam.remote.GetRemoteState(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to fetch remote state: %w", err)
+		}
+
+		// The internal Alertmanager should implement the StateMerger interface.
+		sm := fam.internal.(notifier.StateMerger)
```

Comment:
Let's check that this type assertion succeeds rather than panicking if it doesn't. Best to avoid known panic states where possible.

```suggestion
		sm, ok := fam.internal.(notifier.StateMerger)
		// handle !ok state
```

Reply:
huh, I thought by internalizing this logic we would have access to actual struct and would not have to cast. I wonder if it's not a lot of work to just declare an interface that combines both interfaces notifier.Alertmanager and notifier.StateMerger.